### DISCUSSION TITLE
.env variables in e.g. HTML?

### DISCUSSION THREAD

**Discussion by MarcMenghin** [parcel-bundler/parcel#5666] [Q&A] [answered]
Is there a way to use .env variables to replace e.g. urls in a HTML file? I would like to replace a external JS URL based on node env.

It works fine for JS or TS (using `process.env.ENV_VAR`)

using newest parcel 1.x.

edit: or is it possible with v2?

**Comment by MarcMenghin** [comment, marked as answer]
Was able to use .env with postHTML + posthtml-expressions and a JS configuration.


## .posthtmlrc.js
```js

module.exports = {
	plugins: {
		"posthtml-expressions": {
			locals: {
				"CDN_URL": () => {
					return process.env.CDN_URL;
				}
			}
		}
	}
};

```

## HTML Usage
```html
<script src="{{ CDN_URL }}/latest.js"></script>
```

**Reply by dskvr** [reply]
Tried this solution with an `.env` file and the parameter used returns `undefined`. I also tried to set the ENV before the build command and via an export in bash with the same result. The var was accessible in the parcel script at `process.env`. 

However, renaming `.posthtml.js` to `posthtml.config.js` did the trick!

Edit: This actually didn't work, which trying to get it to work I manually set an `ENV_VAR` and forgot. Once it was unset, it went to not working again. Pretty sure this broke with Parcel V2
